src/stores/user.ts

```diff
@@ -51,26 +51,51 @@ export const useUserStore = defineStore("userStore", {
       }
     },
     async login(credentials: LoginCredentials) {
-      if (this.isLoggedIn) await logout();
-      const response = await login(credentials);
-      const authHeader = response.headers.authorization;
-      const authToken = authHeader.slice("Bearer ".length);
-      setAuthHeader(authHeader);
-      this.setUserData({ authToken, user: response.data });
-      router.push({ name: "HomePage" });
+      try {
+        if (this.isLoggedIn) await logout();
+        const response = await login(credentials);
+        const authHeader = response.headers.authorization;
+        const authToken = authHeader.slice("Bearer ".length);
+        setAuthHeader(authHeader);
+        this.setUserData({ authToken, user: response.data });
+        router.push({ name: "HomePage" });
+      } catch (err) {
+        console.error(err);
+      }
     },
     async register(credentials: RegisterCredentials) {
-      if (this.isLoggedIn) await logout();
-      const response = await register(credentials);
-      const authHeader = response.headers.authorization;
-      const authToken = authHeader.slice("Bearer ".length);
-      setAuthHeader(authHeader);
-      this.setUserData({ authToken, user: response.data });
+      try {
+        if (this.isLoggedIn) await logout();
+        const response = await register(credentials);
+        const authHeader = response.headers.authorization;
+        const authToken = authHeader.slice("Bearer ".length);
+        setAuthHeader(authHeader);
+        this.setUserData({ authToken, user: response.data });
+        router.push({ name: "UserPage" });
+      } catch (err) {
+        console.error(err);
+      }
+    },
+    async getUser() {
+      try {
+        if (this.isLoggedIn) await logout();
```

Comment:
Is there a reason to continue executing this method body? Once we get into that `if` block user will be cleared right? So it will result in some sort of error correct? Instead of that you could throw an `Error` or just return from this method earlier.

Reply:
I am not sure I follow you on this?

Reply:
This method was created to get user data but if user is logged in, you are clearing this user as I assume since `logout` method is called in this block. Wont it result in some sort of error? If yes then maybe it's not necessary to move further with code execution and just stop it here.